src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ExplosionsListener.java

```diff
@@ -41,6 +64,24 @@ public void onEntityExplode(EntityExplodeEvent e) {
 
                     if (blockHandler != null) {
                         success = blockHandler.onBreak(null, block, item, UnregisterReason.EXPLODE);
+                    } else {
+                        item.callItemHandler(BlockBreakHandler.class, handler -> {
+                            if (handler.isExplosionAllowed()) {
+                                BlockStorage.clearBlockInfo(block);
+                                block.setType(Material.AIR);
+
+                                List<ItemStack> drops = new ArrayList<>();
+                                handler.onExplode(block, drops);
```

Comment:
would be great if we could cancel without disabling all explosions. If we had the event we could cancel and then check if cancelled above.

Would be nice for cases like, block durability (for like factions), allows explosion if out of electricity, etc.

Reply:
I added a Block param to the `isExplosionAllowed` method, it will be ignored by default but if you were to implement sth like that, you could override the method and handle durability there.